src/atomic_structure.hh

```diff
@@ -139,141 +158,137 @@ namespace rascal {
         if ((this->pbc.array() != other.pbc.array()).any() or
             (this->cell.array() != other.cell.array()).any() or
+            (this->center_atoms_mask != other.center_atoms_mask).any() or
             (this->positions - other.positions)
                     .rowwise()
                     .squaredNorm()
-                    .maxCoeff() > skin2) {
-          is_similar = false;
+                    .maxCoeff() > threshold2) {
+          is_similar_ = false;
         }
       } else {
-        is_similar = false;
+        is_similar_ = false;
       }
-      return is_similar;
+      return is_similar_;
     }
 
-    inline bool is_identical(const PositionsInput_t & positions,
+    inline bool is_similar(const PositionsInput_t & positions,
+                             const AtomTypesInput_t & atom_types,
+                             const CellInput_t cell, const PBCInput_t & pbc,
+                             const double & threshold2) const {
+      auto center_atoms_mask = ArrayB_t::Ones(atom_types.size());
+      return this->is_similar(positions, atom_types, cell, pbc,
+                                center_atoms_mask, threshold2);
+    }
+
+    inline bool is_similar(const PositionsInput_t & positions,
                              const AtomTypesInput_t & /*atom_types*/,
-                             const Eigen::Ref<const Eigen::MatrixXd> cell,
-                             const PBCInput_t & pbc,
-                             const double & skin2) const {
-      bool is_similar{true};
+                             const CellInput_t cell, const PBCInput_t & pbc,
+                             const ArrayB_ref & center_atoms_mask,
+                             const double & threshold2) const {
+      bool is_similar_{true};
       if (this->positions.cols() == positions.cols()) {
         if ((this->pbc.array() != pbc.array()).any() or
             (this->cell.array() != cell.array()).any() or
+            (this->center_atoms_mask != center_atoms_mask).any() or
             (this->positions - positions).rowwise().squaredNorm().maxCoeff() >
-                skin2) {
-          is_similar = false;
+                threshold2) {
+          is_similar_ = false;
         }
       } else {
-        is_similar = false;
+        is_similar_ = false;
       }
-      return is_similar;
+      return is_similar_;
     }
   };
+
+  /* ---------------------------------------------------------------------- */
+  template <int Dim>
+  void to_json(json & j, const AtomicStructure<Dim> & s) {
+    auto cell = s.cell;
+    cell.transposeInPlace();
+
+    j = json{{"cell", cell},
+             {"atom_types", s.atom_types},
+             {"pbc", s.pbc},
+             {"positions", s.positions},
+             {"center_atoms_mask", s.center_atoms_mask}};
+  }
+
+  /* ---------------------------------------------------------------------- */
+  template <int Dim>
+  void from_json(const json & j, AtomicStructure<Dim> & s) {
```

Comment:
Is there any documentation on the format the JSON must be in, and what all the fields represent? This is related to #65 but even for the atomic structure format the schema and documentation are missing. I know we started with an ASE-like format, but now that we've changed some names and added capabilities (like `center_atoms_mask`), these need to be written down somewhere other than in the source code.

Reply:
it is the ASE json format. center_atoms_mask can be exported with the ase writer if is it in the arrays of the Atoms object under the 'center_atoms_mask' keyword

Reply:
Added an example JSON file that illustrates this behaviour